spec.emu

```diff
@@ -242,31 +250,79 @@ location: https://github.com/tc39/proposal-decimal/
         <dd>It converts a Decimal128 value to a Decimal128 object.</dd>
       </dl>
       <emu-alg>
-        1. Let _O_ be ! OrdinaryCreateFromConstructor(NewTarget, *"%Decimal128.prototype%"*, « [[Decimal128Data]] »).
+        1. Let _O_ be ! OrdinaryCreateFromConstructor(%Decimal128%, *"%Decimal128.prototype%"*, « [[Decimal128Data]] »).
         1. Set _O_.[[Decimal128Data]] to _argument_.
         1. Return _O_.
       </emu-alg>
     </emu-clause>
+
+    <emu-clause id="sec-decimal128-d" type="abstract operation">
+      <h1>
+        EnsureDecimal128Value (
+          _coefficient_ : a mathematical value,
+          _intendedExponent_ : a mathematical value,
+          _roundingMode_: a Decimal128 rounding mode
+        ): either a normal completion containing a Decimal128 value or a throw completion
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>It attempts to convert the pair of arguments, which might be out of range when considered as a Decimal128 value, into a Decimal128 value, throwing an exception if the conversion cannot be done.</dd>
+      </dl>
+      <emu-alg>
+        1. If -6111 ≤ _intendedExponent_ ≤ 6176, then
+          1. If _coefficient_ × 10<sup>_intendedExponent_</sup> is an integer _n_ such that 0 < abs(_n_) < 10<sup>34</sup>, return « _coefficient, _intendedExponent_ ».
+          1. Throw a *RangeError* exception.
+        1. If _intendedExponent_ < 0, return ? EnsureDecimal128Value(_coefficient_ / 10, _intendedExponent_ + 1, _roundingMode_).
+        1. Otherwise, return ? EnsureDecimal128Value(_coefficient_ × 10, _intendedExponent_ + 1, _roundingMode_).
```

Comment:
It'd be a little cleaner if this were written non-recursively, by explicitly doing 10 to the appropriate power